From abd147cf11c2a69381d94e5fda2fecdb8ce1023c Mon Sep 17 00:00:00 2001
From: Lyall Beveridge <lyall.beveridge@morsemicro.com>
Date: Fri, 26 Sep 2025 16:04:24 +1000
Subject: [PATCH] iperf3: update to 3.19.1 (sync with 806ee4b49)

This commit deletes the iPerf3 patches directory
and cherry picks: https://github.com/openwrt/packages/commit/806ee4b4919897a2b1adb1e8aed9bf4d1c25c3e4

Update to latest upstream release.

Changelog: https://github.com/esnet/iperf/releases/tag/3.19.1

Relevant security fixes: E-2025-54349, ESNET-SECADV-2025-0003, CVE-2025-54350,
ESNET-SECADV-2025-0002, VE-2025-54351, ESNET-SECADV-2025-0001

Build system: x86/64
Build-tested: x86/64-glibc
Run-tested: x86/64-glibc

Signed-off-by: Lyall Beveridge <lyall.beveridge@morsemicro.com>
---
 net/iperf3/Makefile                     |  6 +++---
 net/iperf3/patches/010-y2k.patch        | 21 ---------------------
 net/iperf3/patches/020-big-endian.patch | 21 ---------------------
 net/iperf3/patches/030-musl-crash.patch | 19 -------------------
 4 files changed, 3 insertions(+), 64 deletions(-)
 delete mode 100644 net/iperf3/patches/010-y2k.patch
 delete mode 100644 net/iperf3/patches/020-big-endian.patch
 delete mode 100644 net/iperf3/patches/030-musl-crash.patch

diff --git a/feeds/packages/net/iperf3/Makefile b/feeds/packages/net/iperf3/Makefile
index 45d4721bf..c99b8c577 100644
--- a/feeds/packages/net/iperf3/Makefile
+++ b/feeds/packages/net/iperf3/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=iperf
-PKG_VERSION:=3.17.1
-PKG_RELEASE:=4
+PKG_VERSION:=3.19.1
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://downloads.es.net/pub/iperf
-PKG_HASH:=84404ca8431b595e86c473d8f23d8bb102810001f15feaf610effd3b318788aa
+PKG_HASH:=dc63f89ec581ea99f8b558d8eb35109de06383010db5a1906c208a562ba0c270
 
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
 PKG_LICENSE:=BSD-3-Clause
diff --git a/feeds/packages/net/iperf3/patches/010-y2k.patch b/feeds/packages/net/iperf3/patches/010-y2k.patch
deleted file mode 100644
index 86350d1ad..000000000
--- a/feeds/packages/net/iperf3/patches/010-y2k.patch
+++ /dev/null
@@ -1,21 +0,0 @@
-From 4c7629f590bb18855e478a826833adb05d2a128b Mon Sep 17 00:00:00 2001
-From: Rosen Penev <rosenp@gmail.com>
-Date: Sat, 8 Jun 2024 15:35:47 -0700
-Subject: [PATCH] fix -Wformat-y2k error
-
-%c potentially prints a 2 digit year. Just use a normal format.
----
- src/iperf_error.c | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
---- a/src/iperf_error.c
-+++ b/src/iperf_error.c
-@@ -99,7 +99,7 @@ iperf_errexit(struct iperf_test *test, c
-     if (test != NULL && test->timestamps) {
- 	time(&now);
- 	ltm = localtime(&now);
--	strftime(iperf_timestrerr, sizeof(iperf_timestrerr), "%c ", ltm);
-+	strftime(iperf_timestrerr, sizeof(iperf_timestrerr), "%Y-%m-%d %H:%M:%S", ltm);
- 	ct = iperf_timestrerr;
-     }
- 
diff --git a/feeds/packages/net/iperf3/patches/020-big-endian.patch b/feeds/packages/net/iperf3/patches/020-big-endian.patch
deleted file mode 100644
index 35490c9de..000000000
--- a/feeds/packages/net/iperf3/patches/020-big-endian.patch
+++ /dev/null
@@ -1,21 +0,0 @@
-From fe09305eb6f907e4eb637b8edd0c8a986187d1dd Mon Sep 17 00:00:00 2001
-From: Rosen Penev <rosenp@gmail.com>
-Date: Sat, 8 Jun 2024 15:23:51 -0700
-Subject: [PATCH] fix crash under big endian musl
-
-iperf_printf is using an int format here but an int64_t variable. The format only needs the first 3 digits. Cast to int to fix it.
----
- src/iperf_api.c | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
---- a/src/iperf_api.c
-+++ b/src/iperf_api.c
-@@ -4056,7 +4056,7 @@ iperf_print_results(struct iperf_test *t
-                                 iperf_printf(test, report_sender_not_available_summary_format, "SUM");
-                         }
-                         else {
--                          iperf_printf(test, report_sum_bw_retrans_format, mbuf, start_time, sender_time, ubuf, nbuf, total_retransmits, report_sender);
-+                          iperf_printf(test, report_sum_bw_retrans_format, mbuf, start_time, sender_time, ubuf, nbuf, (int)total_retransmits, report_sender);
-                         }
-                 } else {
-                     /* Summary sum, TCP without retransmits. */
diff --git a/feeds/packages/net/iperf3/patches/030-musl-crash.patch b/feeds/packages/net/iperf3/patches/030-musl-crash.patch
deleted file mode 100644
index bc1df68c8..000000000
--- a/feeds/packages/net/iperf3/patches/030-musl-crash.patch
+++ /dev/null
@@ -1,19 +0,0 @@
-From 3da07ae96f5b40f76b75e1ccd4b20267f6a5988e Mon Sep 17 00:00:00 2001
-From: =?UTF-8?q?Jakub=20Kul=C3=ADk?= <kulikjak@gmail.com>
-Date: Wed, 28 Aug 2024 09:43:04 +0200
-Subject: [PATCH] remove incorrect freeaddrinfo call
-
----
- src/net.c | 1 -
- 1 file changed, 1 deletion(-)
-
---- a/src/net.c
-+++ b/src/net.c
-@@ -145,7 +145,6 @@ create_socket(int domain, int proto, con
-     if ((gerror = getaddrinfo(server, portstr, &hints, &server_res)) != 0) {
- 	if (local)
- 	    freeaddrinfo(local_res);
--        freeaddrinfo(server_res);
-         return -1;
-     }
- 
-- 
2.43.0

